guacamole/src/main/webapp/app/auth/service/authenticationService.js

```diff
@@ -290,13 +290,20 @@ angular.module('auth').factory('authenticationService', ['$injector',
         var token = service.getCurrentToken();
         clearAuthenticationResult();
 
-        // Notify listeners that a token is being destroyed
-        $rootScope.$broadcast('guacLogout', token);
+        // Notify listeners that a token is about to be destroyed
+        $rootScope.$broadcast('beforeGuacLogout', token);
```

Comment:
Is there an alternative solution where we have an angular service that components can 'register' handlers in the service.  When someone fires an event through the service it can return a promise that only fires after each handler has been called.

I have been searching for any AngularJS mechanism built in that would accomplish basically 'guaranteed delivery' and I can't find one.  Should be very minimalistic for me to cook this service up if the idea has merit.

Reply:
@knacktim - I think that's a great idea.

Reply:
@necouchman Here ya go! https://github.com/apache/guacamole-client/pull/573